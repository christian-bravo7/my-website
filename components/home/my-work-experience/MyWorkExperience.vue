<template>
  <section class="section-container">
    <div class="max-w-screen-lg | flex justify-center items-center | my-0 mx-auto | overflow-x-hidden">
      <div class="flex flex-col | w-full">
        <div class="flex justify-center | mb-16">
          <h2 class="section-title my-work-experience__title">
            {{ $t('home.where-i-have-worked') }}
          </h2>
        </div>
        <div class="flex flex-col md:flex-row">
          <div class="relative | my-work-experience__button-wrapper">
            <div class="flex md:flex-col | mb-8 md:mb-0 pb-4 md:pb-0 | overflow-x-scroll md:overflow-x-auto | my-work-experience__button-list">
              <MyWorkExperienceButton
                :is-active="currentTab === 'telus'"
                @onClick="changeTab('telus')"
              >
                Telus International
              </MyWorkExperienceButton>
              <MyWorkExperienceButton
                :is-active="currentTab === 'master'"
                @onClick="changeTab('master')"
              >
                Master Development
              </MyWorkExperienceButton>
              <MyWorkExperienceButton
                :is-active="currentTab === 'viaro'"
                @onClick="changeTab('viaro')"
              >
                Viaro Networks
              </MyWorkExperienceButton>
            </div>
          </div>
          <div class="flex-1 | md:ml-6">
            <MyWorkExperienceCard v-bind="workExperienceData[currentTab]" />
          </div>
        </div>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class MyWorkExperience extends Vue {
  currentTab: string = 'telus';

  get workExperienceData (): any {
    return {
      telus: {
        jobPosition: 'Software Developer',
        company: 'Telus International',
        websiteUrl: 'https://www.telusinternational.com',
        startYear: [2020, 11],
        descriptionParagraphs: this.$i18n.t('my-work-experience.masterdevel.description'),
        techStack: ['Angular', 'Typescript', 'Javascript', 'jQuery', 'i18n', 'Java Portlets', 'LESS'],
      },
      master: {
        jobPosition: 'Frontend Developer',
        company: 'Master Development',
        websiteUrl: 'https://www.linkedin.com/company/corporaci%C3%B3n-master-group',
        startYear: [2019, 9],
        endYear: [2020, 7],
        descriptionParagraphs: this.$i18n.t('my-work-experience.masterdevel.description'),
        techStack: ['NuxtJS', 'Vue', 'Vuex', 'Angular', 'Typescript', 'i18n', 'Bulma', 'SCSS', 'Pug', 'Handlebars', 'Prismic CMS', 'NodeJS'],
      },
      viaro: {
        jobPosition: 'Quality Assurance',
        company: 'Viaro Networks',
        websiteUrl: 'https://www.viaro.net',
        startYear: [2018, 5],
        descriptionParagraphs: this.$i18n.t('my-work-experience.viaro.description'),
        endYear: [2019, 9],
        tools: ['BrowserStack', 'Jira', 'Lighthouse by Google'],
      },
    };
  }

  changeTab (nextTab: string): void {
    this.currentTab = nextTab;
  }
}
</script>

<style lang="scss" scoped>
.my-work-experience {
  &__title {
    @include theme-gradients(four, five);
  }

  &__button-list {
    &::after {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      width: 80px;
      background: linear-gradient(90deg, rgba(255, 255, 255, 0) 0%, rgba(255, 255, 255, 1) 70%);
      content: '';
      pointer-events: none;
    }
  }
}

@screen md {
  .my-work-experience {
    &__button-wrapper {
      width: 210px;
    }

    &__button-list {
      &::after {
        display: none;
      }
    }
  }
}
</style>
