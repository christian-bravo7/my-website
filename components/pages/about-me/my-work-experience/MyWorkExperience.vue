<template>
  <section class="section-container">
    <div class="max-w-screen-lg | flex justify-center items-center | w-full | my-0 mx-auto | overflow-x-hidden">
      <div class="flex flex-col | w-full">
        <div class="flex justify-center | mb-16">
          <h2 class="section-title my-work-experience__title">
            {{ $t('home.where-i-have-worked') }}
          </h2>
        </div>
        <div class="relative flex flex-col md:flex-row">
          <div class="my-work-experience__button-wrapper">
            <div class="relative | flex md:flex-col | mb-8 md:mb-0 pb-4 md:pb-0 | overflow-x-scroll md:overflow-x-auto | my-work-experience__button-list">
              <MyWorkExperienceButton
                v-for="(work, key) in workExperienceData"
                :id="key"
                :key="key"
                class="mr-1 md:ml-1 md:mr-0 | my-work-experience__button"
                :is-active="currentTab === key"
                @click="changeTab(key)"
              >
                {{ work.company }}
              </MyWorkExperienceButton>
              <div class="absolute | left-0 bottom-3 md:bottom-0 h-1 md:h-full w-full md:w-1 | rounded-full | bg-gray-100 dark:bg-blue-700">
                <span
                  class="block | absolute | md:w-full h-full md:h-14 | rounded-full | bg-pink-500 dark:bg-blue-400 | my-work-experience__button-marker"
                  :style="{ 'top': `${markTopPosition}px`, 'left': `${markLeftPosition}px`, 'width': `${markWidth}px` }"
                />
              </div>
            </div>
            <span class="my-work-experience__button-list-fade" />
          </div>
          <div class="flex-1 | md:ml-6 | my-work-experience__card-wrapper">
            <MyWorkExperienceCard v-bind="workExperienceData[currentTab]" />
          </div>
        </div>
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component } from 'nuxt-property-decorator';

@Component
export default class MyWorkExperience extends Vue {
  currentTab: string = 'telus';
  markTopPosition: number = 0;
  markLeftPosition: number = 0;
  markWidth: number = 0;

  get workExperienceData (): any {
    return {
      telus: {
        jobPosition: 'Software Developer',
        company: 'TELUS International',
        websiteUrl: 'https://www.telusinternational.com',
        startYear: [2020, 11],
        descriptionParagraphs: this.$i18n.t('my-work-experience.telus.description'),
        techStack: ['Angular', 'Typescript', 'Javascript', 'jQuery', 'i18n', 'Java (Liferay)', 'SCSS', 'LESS'],
      },
      master: {
        jobPosition: 'Frontend Developer',
        company: 'Master Development',
        websiteUrl: 'https://www.linkedin.com/company/corporaci%C3%B3n-master-group',
        startYear: [2019, 9],
        endYear: [2020, 7],
        descriptionParagraphs: this.$i18n.t('my-work-experience.masterdevel.description'),
        techStack: ['NuxtJS', 'Vue', 'Vuex', 'Angular', 'Typescript', 'i18n', 'Bulma', 'SCSS', 'Pug', 'Handlebars', 'Prismic CMS', 'NodeJS'],
      },
      viaro: {
        jobPosition: 'Quality Assurance',
        company: 'Viaro Networks',
        websiteUrl: 'https://www.viaro.net',
        startYear: [2018, 5],
        descriptionParagraphs: this.$i18n.t('my-work-experience.viaro.description'),
        endYear: [2019, 9],
        tools: ['BrowserStack', 'Jira', 'Lighthouse by Google'],
      },
    };
  }

  changeTab (nextTab: string): void {
    const element = document.getElementById(nextTab) as HTMLElement;

    this.currentTab = nextTab;
    this.markTopPosition = element.offsetTop;
    this.markLeftPosition = element.offsetLeft;
    this.markWidth = element.offsetWidth;
  }
}
</script>

<style lang="scss" scoped>
.my-work-experience {
  &__title {
    @include theme-gradients(four, five);
  }

  &__button-list {
    &::after {
      display: block;
      height: 20px;
      margin-left: 40px;
      padding: 4px;
      content: '';
    }
  }

  &__button {
    min-width: 210px;
  }

  &__button-marker {
    top: initial;
    transition-delay: 100ms;
    transition-duration: 300ms;
    transition-property: width, left;
  }

  &__button-list-fade {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    z-index: 10;
    width: 100px;
    pointer-events: none;
    content: '';

    @apply bg-gradient-to-r from-transparent to-gray-50 dark:to-blue-800;
  }
}

@screen md {
  .my-work-experience {
    &__button-wrapper {
      width: 210px;
    }

    &__button {
      min-width: auto;
    }

    &__button-list-fade {
      display: none;
    }

    &__button-marker {
      left: 0 !important;
      width: 100% !important;
      transition-property: top;
    }

    &__button-list {
      &::after {
        display: none;
      }
    }

    &__card-wrapper {
      min-height: 430px;
    }
  }
}
</style>
