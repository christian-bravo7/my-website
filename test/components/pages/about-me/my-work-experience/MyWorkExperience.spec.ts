import { createLocalVue, mount } from '@vue/test-utils';

import MyWorkExperience from '@/components/pages/about-me/my-work-experience/MyWorkExperience.vue';

import setupI18n from '@/test/i18n';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);

const componentConfig = {
  localVue,
  i18n,
};

describe('MyWorkExperience component', () => {
  it('should render correclty', () => {
    const wrapper = mount(MyWorkExperience, {
      ...componentConfig,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
