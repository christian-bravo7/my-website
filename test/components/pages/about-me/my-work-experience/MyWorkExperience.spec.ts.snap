// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyWorkExperience component should render correclty 1`] = `
<section class="section-container">
  <div class="max-w-screen-lg | flex justify-center items-center | w-full | my-0 mx-auto | overflow-x-hidden">
    <div class="flex flex-col | w-full">
      <div class="flex justify-center | mb-16">
        <h2 class="section-title my-work-experience__title">
          Where I've worked
        </h2>
      </div>
      <div class="relative flex flex-col md:flex-row">
        <div class="my-work-experience__button-wrapper">
          <div class="relative | flex md:flex-col | mb-8 md:mb-0 pb-4 md:pb-0 | overflow-x-scroll md:overflow-x-auto | my-work-experience__button-list"><button id="telus" class="relative | px-4 py-4 | text-center md:text-left | rounded-lg md:rounded-tl-none md:rounded-bl-none | outline-none focus:outline-none | text-black dark:text-white | whitespace-nowrap | transition-all duration-300 | my-work-experience-button mr-1 md:ml-1 md:mr-0 | my-work-experience__button my-work-experience-button--is-active">
              TELUS International
            </button><button id="master" class="relative | px-4 py-4 | text-center md:text-left | rounded-lg md:rounded-tl-none md:rounded-bl-none | outline-none focus:outline-none | text-black dark:text-white | whitespace-nowrap | transition-all duration-300 | my-work-experience-button mr-1 md:ml-1 md:mr-0 | my-work-experience__button">
              Master Development
            </button><button id="viaro" class="relative | px-4 py-4 | text-center md:text-left | rounded-lg md:rounded-tl-none md:rounded-bl-none | outline-none focus:outline-none | text-black dark:text-white | whitespace-nowrap | transition-all duration-300 | my-work-experience-button mr-1 md:ml-1 md:mr-0 | my-work-experience__button">
              Viaro Networks
            </button>
            <div class="absolute | left-0 bottom-3 md:bottom-0 h-1 md:h-full w-full md:w-1 | rounded-full | bg-gray-100 dark:bg-blue-700"><span class="block | absolute | md:w-full h-full md:h-14 | rounded-full | bg-pink-500 dark:bg-blue-400 | my-work-experience__button-marker" style="top: 0px; left: 0px; width: 0px;"></span></div>
          </div> <span class="my-work-experience__button-list-fade"></span>
        </div>
        <div class="flex-1 | md:ml-6 | my-work-experience__card-wrapper">
          <div>
            <h6 class="text-xl | mb-2"><span class="font-bold">
    Software Developer
  </span> <span>at</span> <a href="https://www.telusinternational.com" target="_blank" rel="noopener noreferrer" class="text-pink-500 dark:text-blue-400 | font-bold | hover:underline | transition-all duration-300">
                TELUS International
              </a></h6>
            <div class="flex text-sm mb-6"><span><span class="capitalize">
      Nov 2020
    </span> <span>-</span> <span class="capitalize">
      present
    </span></span> <span class="flex items-center | mx-2"><span class="w-1 h-1 | rounded-full | bg-black dark:bg-white"></span></span> <span class="text-gray-700 dark:text-gray-300">
    10 months
  </span></div>
            <div class="mb-6">
              <div class="flex mb-2"><span class="text-pink-600 dark:text-blue-300 | material-icons">
      arrow_right
    </span> <span class="leading-loose | font-base">
      Development of new features, bug fixing and support for web applications.
    </span></div>
            </div>
            <div><span class="block | font-bold | text-xs | mb-6">
    Technologies:
  </span>
              <div class="flex flex-wrap">
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  Angular
                </div>
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  Typescript
                </div>
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  Javascript
                </div>
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  jQuery
                </div>
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  i18n
                </div>
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  Java (Liferay)
                </div>
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  SCSS
                </div>
                <div class="mr-4 mb-4 py-2 px-4 | text-sm | font-bold | whitespace-nowrap | rounded-full | border border-pink-500 dark:border-blue-300 | bg-pink-50 bg-opacity-30 dark:bg-blue-700 dark:bg-opacity-25">
                  LESS
                </div>
              </div>
            </div>
            <!---->
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
